pkg/webhook/admission/webhook.go

```diff
@@ -253,3 +253,21 @@ func StandaloneWebhook(hook *Webhook, opts StandaloneOptions) (http.Handler, err
 	}
 	return metrics.InstrumentedHook(opts.MetricsPath, hook), nil
 }
+
+// requestContextKey is how we find the admission.Request in a context.Context.
+type requestContextKey struct{}
+
+// RequestFromContext returns an admission.Request from ctx.
+func RequestFromContext(ctx context.Context) (Request, error) {
```

Comment:
It would be a nicer UX if we could omit the error return parameter. But returning an empty Request if it is not found feels wrong

Reply:
+1, an error return parameter will be nicer, even if that should not happen for custom validator/defaulter.

Reply:
The error return message seems ok, better to have the clarity rather than checking nils?

Reply:
Yup and a lot better then checking empty fields of an empty struct (as it's not a pointer)